hal/halx86/apic/halinit.c

```diff
@@ -24,6 +25,15 @@ HalpInitProcessor(
     IN ULONG ProcessorNumber,
     IN PLOADER_PARAMETER_BLOCK LoaderBlock)
 {
+    if (ProcessorNumber == 0)
```

Comment:
This should always be TRUE when `CONFIG_SMP` is not defined?

Reply:
It should always be true when it's not defined yeah, but this function gets called numerous times when using SMP of course too.

Reply:
My point being here, that this if-test (NOT its code block) could be surrounded by a `#ifdef CONFIG_SMP`, I think 🤔 (i.e. disabling the test in monoprocessor build and run the code block unconditionally.)